h5cpp does not compile with hdf5 >= 1.13

Starting from hdf5 1.13.0 `hsize_t` was changed from
`long long unsigned int` to `uint64_t` therefore we can see
```
[ 22%] Building CXX object src/h5cpp/CMakeFiles/h5cpp.dir/node/dataset.cpp.o
[ 22%] Building CXX object src/h5cpp/CMakeFiles/h5cpp.dir/node/functions.cpp.o
/__w/h5cpp/h5cpp/src/h5cpp/node/dataset.cpp: In member function 'long long unsigned int hdf5::node::Dataset::chunk_storage_size(std::vector<long long unsigned int>) const':
/__w/h5cpp/h5cpp/src/h5cpp/node/dataset.cpp:199:43: error: invalid conversion from 'long long unsigned int*' to 'const hsize_t*' {aka 'const long unsigned int*'} [-fpermissive]
  199 |                                offset.data(),
      |                                ~~~~~~~~~~~^~
      |                                           |
      |                                           long long unsigned int*
In file included from /github/home/.conan2/p/hdf5a21fda4faa24a/p/include/hdf5/hdf5.h:24,
                 from /__w/h5cpp/h5cpp/src/h5cpp/core/hdf5_capi.hpp:44,
                 from /__w/h5cpp/h5cpp/src/h5cpp/core/object_handle.hpp:30,
                 from /__w/h5cpp/h5cpp/src/h5cpp/node/node.hpp:28,
                 from /__w/h5cpp/h5cpp/src/h5cpp/node/dataset.hpp:30,
                 from /__w/h5cpp/h5cpp/src/h5cpp/node/dataset.cpp:32:
/github/home/.conan2/p/hdf5a21fda4faa24a/p/include/hdf5/H5Dpublic.h:606:71: note:   initializing argument 2 of 'herr_t H5Dget_chunk_storage_size(hid_t, const hsize_t*, hsize_t*)'
  606 | H5_DLL herr_t H5Dget_chunk_storage_size(hid_t dset_id, const hsize_t *offset, hsize_t *chunk_bytes);
      |                                                        ~~~~~~~~~~~~~~~^~~~~~
/__w/h5cpp/h5cpp/src/h5cpp/node/dataset.cpp:200:32: error: invalid conversion from 'long long unsigned int*' to 'hsize_t*' {aka 'long unsigned int*'} [-fpermissive]
  200 |                                &storage_size)<0)
      |                                ^~~~~~~~~~~~~
      |                                |
      |                                long long unsigned int*
In file included from /github/home/.conan2/p/hdf5a21fda4faa24a/p/include/hdf5/hdf5.h:24,
                 from /__w/h5cpp/h5cpp/src/h5cpp/core/hdf5_capi.hpp:44,
                 from /__w/h5cpp/h5cpp/src/h5cpp/core/object_handle.hpp:30,
                 from /__w/h5cpp/h5cpp/src/h5cpp/node/node.hpp:28,
                 from /__w/h5cpp/h5cpp/src/h5cpp/node/dataset.hpp:30,
                 from /__w/h5cpp/h5cpp/src/h5cpp/node/dataset.cpp:32:
/github/home/.conan2/p/hdf5a21fda4faa24a/p/include/hdf5/H5Dpublic.h:606:88: note:   initializing argument 3 of 'herr_t H5Dget_chunk_storage_size(hid_t, const hsize_t*, hsize_t*)'
  606 | H5_DLL herr_t H5Dget_chunk_storage_size(hid_t dset_id, const hsize_t *offset, hsize_t *chunk_bytes);
      |                                                                               ~~~~~~~~~^~~~~~~~~~~
src/h5cpp/CMakeFiles/h5cpp.dir/build.make:647: recipe for target 'src/h5cpp/CMakeFiles/h5cpp.dir/node/dataset.cpp.o' failed
make[2]: *** [src/h5cpp/CMakeFiles/h5cpp.dir/node/dataset.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:173: recipe for target 'src/h5cpp/CMakeFiles/h5cpp.dir/all' failed
make[1]: *** [src/h5cpp/CMakeFiles/h5cpp.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/conanfile.py
+++ b/conanfile.py
@@ -32,7 +32,7 @@ class H5CppConan(ConanFile):
             self.options["hdf5"].parallel = True
 
     def requirements(self):
-        self.requires("hdf5/1.12.2")
+        self.requires("hdf5/1.14.0")
         if self.options.get_safe("with_boost", False):
             if self.settings.os == "Windows":
                 self.requires("boost/1.77.0")

--- a/src/h5cpp/core/object_handle.cpp
+++ b/src/h5cpp/core/object_handle.cpp
@@ -303,7 +303,7 @@ ObjectHandle::Type ObjectHandle::get_type() const
 #else
     case H5I_REFERENCE:
 #endif
-#if H5_VERSION_GE(1,14,0)
+#if H5_VERSION_GE(1,13,0)
     case H5I_EVENTSET:
 #endif
     case H5I_NTYPES:

--- a/src/h5cpp/core/object_handle.cpp
+++ b/src/h5cpp/core/object_handle.cpp
@@ -303,6 +303,9 @@ ObjectHandle::Type ObjectHandle::get_type() const
 #else
     case H5I_REFERENCE:
 #endif
+#if H5_VERSION_GE(1,14,0)
+    case H5I_EVENTSET:
+#endif
     case H5I_NTYPES:
       break;
   }

--- a/src/h5cpp/file/file.hpp
+++ b/src/h5cpp/file/file.hpp
@@ -221,7 +221,7 @@ size_t File::to_buffer_reshape(T &data,
 			       const datatype::Datatype &mem_type,
 			       const dataspace::Dataspace &mem_space) const
 {
-  size_t databytesize = signed2unsigned<unsigned long long>(mem_space.size()) * mem_type.size();
+  size_t databytesize = signed2unsigned<hsize_t>(mem_space.size()) * mem_type.size();
   ssize_t s = 0;
   if(mem_type.get_class() == datatype::Class::Integer)
     {

--- a/src/h5cpp/node/dataset.cpp
+++ b/src/h5cpp/node/dataset.cpp
@@ -191,10 +191,10 @@ void Dataset::refresh() const
 
 #if H5_VERSION_GE(1,10,2)
 
-unsigned long long Dataset::chunk_storage_size(
-			 std::vector<unsigned long long> offset) const
+hsize_t Dataset::chunk_storage_size(
+			 std::vector<hsize_t> offset) const
 {
-  unsigned long long storage_size;
+  hsize_t storage_size;
   if(H5Dget_chunk_storage_size(static_cast<hid_t>(*this),
 			       offset.data(),
 			       &storage_size)<0)
@@ -258,9 +258,9 @@ void resize_by(const Dataset &dataset,size_t dimension_index,ssize_t delta)
     throw std::runtime_error(ss.str());
   }
   if (delta < 0)
-    current_dims[dimension_index] -= signed2unsigned<unsigned long long>(-delta);
+    current_dims[dimension_index] -= signed2unsigned<hsize_t>(-delta);
   else
-    current_dims[dimension_index] += signed2unsigned<unsigned long long>(delta);
+    current_dims[dimension_index] += signed2unsigned<hsize_t>(delta);
   dataset.resize(current_dims);
 }
 

--- a/src/h5cpp/node/dataset.hpp
+++ b/src/h5cpp/node/dataset.hpp
@@ -338,14 +338,14 @@ class DLL_EXPORT Dataset : public Node
     //!
     template<typename T>
     void write_chunk(const T &data,
-		     std::vector<unsigned long long> offset,
+		     std::vector<hsize_t> offset,
 		     std::uint32_t filter_mask = 0,
 		     const property::DatasetTransferList &dtpl =
 		     property::DatasetTransferList::get());
 
     template<typename T>
     void write_chunk(const T &data,
-		     std::vector<unsigned long long> offset,
+		     std::vector<hsize_t> offset,
 		     std::uint32_t filter_mask = 0,
 		     const property::DatasetTransferList &dtpl =
 		     property::DatasetTransferList::get()) const;
@@ -368,7 +368,7 @@ class DLL_EXPORT Dataset : public Node
     void write_chunk(const T &data,
 		     const datatype::Datatype &mem_type,
 		     const dataspace::Dataspace &mem_space,
-		     std::vector<unsigned long long> & offset,
+		     std::vector<hsize_t> & offset,
 		     std::uint32_t filter_mask = 0,
 		     const property::DatasetTransferList &dtpl =
 		     property::DatasetTransferList::get())  const;
@@ -389,7 +389,7 @@ class DLL_EXPORT Dataset : public Node
     //!
     template<typename T>
       std::uint32_t read_chunk(T &data,
-			       std::vector<unsigned long long> offset,
+			       std::vector<hsize_t> offset,
 			       const property::DatasetTransferList &dtpl =
 			       property::DatasetTransferList::get())  const;
 
@@ -410,7 +410,7 @@ class DLL_EXPORT Dataset : public Node
     template<typename T>
       std::uint32_t read_chunk(T &data,
 			       const datatype::Datatype &mem_type,
-			       std::vector<unsigned long long> & offset,
+			       std::vector<hsize_t> & offset,
 			       const property::DatasetTransferList &dtpl =
 			       property::DatasetTransferList::get())  const;
 
@@ -424,7 +424,7 @@ class DLL_EXPORT Dataset : public Node
     //! \param offset logical position of the first element of the chunk in the dataset's dataspace
     //! \return the size in bytes for the chunk.
     //!
-      unsigned long long chunk_storage_size(std::vector<unsigned long long> offset)  const;
+      hsize_t chunk_storage_size(std::vector<hsize_t> offset)  const;
 
 #endif
 
@@ -863,7 +863,7 @@ void Dataset::write(const T &data,
 
 template<typename T>
 void Dataset::write_chunk(const T &data,
-                          std::vector<unsigned long long> offset,
+                          std::vector<hsize_t> offset,
                           std::uint32_t filter_mask,
                           const property::DatasetTransferList &dtpl)
 {
@@ -874,7 +874,7 @@ void Dataset::write_chunk(const T &data,
 
 template<typename T>
 void Dataset::write_chunk(const T &data,
-                          std::vector<unsigned long long> offset,
+                          std::vector<hsize_t> offset,
                           std::uint32_t filter_mask,
                           const property::DatasetTransferList &dtpl) const
 {
@@ -887,11 +887,11 @@ template<typename T>
 void Dataset::write_chunk(const T &data,
 			  const datatype::Datatype &mem_type,
 			  const dataspace::Dataspace &mem_space,
-			  std::vector<unsigned long long> & offset,
+			  std::vector<hsize_t> & offset,
                           std::uint32_t filter_mask,
                           const property::DatasetTransferList &dtpl) const
 {
-  size_t databytesize = signed2unsigned<unsigned long long>(mem_space.size()) * mem_type.size();
+  size_t databytesize = signed2unsigned<hsize_t>(mem_space.size()) * mem_type.size();
 
   if(mem_type.get_class() == datatype::Class::Integer)
     {
@@ -933,7 +933,7 @@ void Dataset::write_chunk(const T &data,
 
 template<typename T>
 std::uint32_t Dataset::read_chunk(T &data,
-			 std::vector<unsigned long long> offset,
+			 std::vector<hsize_t> offset,
 			 const property::DatasetTransferList &dtpl) const
 {
   hdf5::datatype::DatatypeHolder mem_type_holder;
@@ -943,7 +943,7 @@ std::uint32_t Dataset::read_chunk(T &data,
 template<typename T>
 std::uint32_t Dataset::read_chunk(T &data,
 				  const datatype::Datatype &mem_type,
-				  std::vector<unsigned long long> & offset,
+				  std::vector<hsize_t> & offset,
 				  const property::DatasetTransferList &dtpl) const
 {
   std::uint32_t filter_mask;

--- a/test/error/descriptor_test.cpp
+++ b/test/error/descriptor_test.cpp
@@ -48,7 +48,7 @@ SCENARIO("extracting error data") {
         }
         THEN("we can extract the error strings") {
           d.extract_strings();
-#if H5_VERSION_GE(1,14,0)
+#if H5_VERSION_GE(1,13,0)
           REQUIRE(d.major == "Object ID");
 #else
           REQUIRE(d.major == "Object atom");

--- a/test/error/descriptor_test.cpp
+++ b/test/error/descriptor_test.cpp
@@ -48,7 +48,11 @@ SCENARIO("extracting error data") {
         }
         THEN("we can extract the error strings") {
           d.extract_strings();
+#if H5_VERSION_GE(1,14,0)
+          REQUIRE(d.major == "Object ID");
+#else
           REQUIRE(d.major == "Object atom");
+#endif
           REQUIRE(d.minor == "Can't get value");
         }
       }

